src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -281,7 +281,7 @@ const TopicFragment = ({
           data={data}
           renderItem={renderItem}
           keyExtractor={(item, index) => index.toString()}
-          onEndReached={() => fetchData()}
+          onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good one as it adds a check to ensure that `fetchData()` is only called when `firstData` has elements. This can prevent unnecessary network calls and improve performance. However, it would be better to use a more explicit check for array length.

```diff
- onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
+ onEndReached={() => (firstData && firstData.length > 0 ? fetchData() : null)}
```
This additional check ensures that `firstData` is not `null` or `undefined` before checking its length, which could prevent potential runtime errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->